src/frecent_paths.rs

```diff
@@ -148,6 +149,18 @@ impl PathFrecency {
             &pc_ci_sm,
         ];
 
+        // If the user has typed an exact path,
+        // add it to the database
+        env::current_dir().map(|mut cwd| {
+            cwd.push(filter);
```

Comment:
I assume the behaviour of `push` overwriting if `filter` is absolute is known and intended here. 
It does seem like the right behaviour.

Reply:
Yep, that was the intention!